algorithms/linfa-ftrl/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+#![doc = include_str!("../README.md")]
+
+mod algorithm;
+mod error;
+mod hyperparams;
+
+use crate::hyperparams::FtrlParams;
+pub use algorithm::Result;
+pub use error::FtrlError;
+use linfa::Float;
+use ndarray::Array1;
+use ndarray_rand::RandomExt;
+use rand::distributions::Uniform;
+
+#[cfg(feature = "serde")]
```

Comment:
Feature-gating by `serde` does nothing here, because you don't have a `serde` feature in this crate. And there's no point in having a `serde` feature because `argmin` doesn't feature-gate its dependency on `serde`.